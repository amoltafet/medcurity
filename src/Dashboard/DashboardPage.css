.dashboard {
  margin-bottom: 1rem;
  margin-top: 1rem;
  margin-left: 2rem;
  margin-right: 2rem;

}

