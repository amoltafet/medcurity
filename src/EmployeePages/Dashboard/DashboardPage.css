.dashboard {
  margin-bottom: 1%;
  margin-top: 1%;
  margin-left: 2%;
  margin-right: 2%;

}
.dashTopPanel {
  height: 38%;
}
