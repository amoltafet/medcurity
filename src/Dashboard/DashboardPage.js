import React from 'react';
import './DashboardPage.css';
import 'bootstrap/dist/css/bootstrap.min.css'
import { CardDeck } from 'react-bootstrap'
import WelcomePanel from './WelcomePanel';
import Leaderboard from './DashLeaderboard';
import MenuBar from '../MenuBar/MenuBar';
import { useEffect, useState } from 'react';
import LearningModulesCards from './LearningModulesCards';
import LearningModulesDirectories from './LearningModuleDirectories';
import Axios from 'axios';


/**
* Creates and holds all of the componets for the Dashboard. 
* @return {DashboardPage}
*/
const DashboardPage = () => {
    Axios.defaults.withCredentials = true;
    const [session, setSession] = useState([]);

    useEffect(() => {
        Axios.get("http://localhost:3002/users/login").then((response) => {
          console.log('aaahhh', response.data.user)
          setSession(response.data.user[0])
        }).catch(error => console.error(`Error ${error}`));
      }, []);

    console.log(session)

  return (
    <>
		<MenuBar></MenuBar>
		<CardDeck className="dashTopPanel" style={{display: 'flex', flexDirection: 'row'}}>
			<WelcomePanel user={session} ></WelcomePanel> 
			<Leaderboard user={session} ></Leaderboard>
        </CardDeck>
        <LearningModulesCards user={session} />
        <LearningModulesDirectories/>
    </>
  );
}

export default DashboardPage;