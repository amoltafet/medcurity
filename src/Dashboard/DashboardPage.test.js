import renderer from 'react-test-renderer';
import DashboardPage from './DashboardPage';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

test('renders DashboardPage', () => {
  const tree = renderer.create(<BrowserRouter><DashboardPage /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
