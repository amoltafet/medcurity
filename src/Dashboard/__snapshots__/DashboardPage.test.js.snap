// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DashboardPage 1`] = `
Array [
  <img
    alt=""
    className="MedcurityLogo"
    src="/Medcurity_Logo.png"
    variant="top"
  />,
  <form
    className="invalid_bodyForm img-fluid"
  >
    <div
      className="invalid_bodyCol rounded col"
    >
      <div
        className="invalid_headerRow row"
      >
        <img
          alt=""
          className="invalid_lockImg"
          src="/invalid_lock.png"
          variant="top"
        />
        <h2
          className="invalid_h1"
        >
          Oops! We're not sure how you got here, but...
        </h2>
        <div
          className="invalid_messageDiv"
        >
          <h5
            className="invalid_h5"
          >
            You need to be logged in to view your dashboard.
          </h5>
        </div>
        <button
          className="invalid_redirectButton btn btn-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Back to Login Page
        </button>
      </div>
    </div>
  </form>,
]
`;
